heft-plugins/heft-lint-plugin/src/SarifFormatter.ts

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable unicorn/no-null */
+/**
+ * @fileoverview SARIF v2.1 formatter
+ * @author Microsoft
+ */
+
+import url from 'url';
+
+import type * as TEslint from 'eslint';
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
```

Comment:
Nit: these section headers are pretty unused in this repo, not sure they're needed here